File: 1695-maximum-unique-subarray/solution.go
/*
自己完成
*/
package main
import "fmt"
func maximumUniqueSubarray(nums []int) int {
	numsMap := make(map[int]int)
	score, maxScore := 0, 0
    l, r := 0, 0
	for r < len(nums){
		numsMap[nums[r]]++
		score += nums[r]
		for numsMap[nums[r]] > 1{
			numsMap[nums[l]]--
			score -= nums[l]
			l++
		}
		maxScore = max(maxScore, score)
		r++
	}
	return  maxScore
}
func main(){
	nums := []int {5,2,1,2,5,2,1,2,5}
	fmt.Println("Ans = ", maximumUniqueSubarray(nums))
}
/*
執行代碼:
go run ./1695-maximum-unique-subarray
*/

File: 1695-maximum-unique-subarray/solution.py
"""
自己完成
"""
from typing import List
from collections import defaultdict
class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        l, r = 0, 0
        score, max_score = 0, 0
        dd = defaultdict(int)
        while r < len(nums):
            dd[nums[r]] += 1
            score += nums[r]
            while dd[nums[r]] > 1:
                dd[nums[l]] -= 1
                score -= nums[l]
                l += 1
            max_score = max(max_score, score)
            r += 1
        return max_score
if __name__ == "__main__":
    nums = [5]
    print("Ans = ", Solution().maximumUniqueSubarray(nums))